subsys/bluetooth/host/conn.c

```diff
@@ -898,6 +898,22 @@ __maybe_unused static bool dont_have_methods(struct bt_conn *conn)
 		(conn->has_data == NULL);
 }
 
+static void move_head_to_tail(void)
```

Comment:
Should calls to this function, or at least the slist access, but guarded by k_sched_lock? 

(reference to https://github.com/zephyrproject-rtos/zephyr/pull/83700)